cuckoo/processing/static.py

```diff
@@ -250,45 +251,61 @@ def _get_signature(self):
         if not dir_entry or not dir_entry.VirtualAddress or not dir_entry.Size:
             return []
 
-        if not HAVE_MCRYPTO:
-            log.critical(
-                "You do not have the m2crypto library installed preventing "
-                "certificate extraction. Please read the Cuckoo "
-                "documentation on installing m2crypto (you need SWIG "
-                "installed and then `pip install m2crypto==0.24.0`)!"
-            )
-            return []
-
         signatures = self.pe.write()[dir_entry.VirtualAddress+8:]
-        bio = M2Crypto.BIO.MemoryBuffer(signatures)
+        bio = backend._bytes_to_bio(signatures)
         if not bio:
             return []
 
-        pkcs7_obj = M2Crypto.m2.pkcs7_read_bio_der(bio.bio_ptr())
+        pkcs7_obj = backend._lib.d2i_PKCS7_bio(bio.bio, backend._ffi.NULL)
         if not pkcs7_obj:
             return []
 
         ret = []
-        p7 = M2Crypto.SMIME.PKCS7(pkcs7_obj)
-        for cert in p7.get0_signers(M2Crypto.X509.X509_Stack()) or []:
-            subject = cert.get_subject()
-            ret.append({
-                "serial_number": "%032x" % cert.get_serial_number(),
-                "common_name": subject.CN,
-                "country": subject.C,
-                "locality": subject.L,
-                "organization": subject.O,
-                "email": subject.Email,
-                "sha1": "%040x" % int(cert.get_fingerprint("sha1"), 16),
-                "md5": "%032x" % int(cert.get_fingerprint("md5"), 16),
-            })
-
-            if subject.GN and subject.SN:
-                ret[-1]["full_name"] = "%s %s" % (subject.GN, subject.SN)
-            elif subject.GN:
-                ret[-1]["full_name"] = subject.GN
-            elif subject.SN:
-                ret[-1]["full_name"] = subject.SN
+        signers = backend._lib.PKCS7_get0_signers(pkcs7_obj, backend._ffi.NULL, 0)
```

Comment:
This does not work with enveloped digital certificates. Even though you are running with a for loop, there will always be only one certificate as a result from PKCS7_get0_signers. I haven' get it running with crypotgraphy, but I have with pyopenssl.
`from OpenSSL import crypto

    certs = crypto._ffi.NULL
    pkcs7 = crypto.load_pkcs7_data(
        crypto.FILETYPE_ASN1,
        extract_signature(pe)
    )

    if pkcs7.type_is_signed():
        certs = pkcs7._pkcs7.d.sign.cert
    elif pkcs7.type_is_signedAndEnveloped():
        certs = pkcs7._pkcs7.d.signed_and_enveloped.cert

    pycerts = []
    for i in range(crypto._lib.sk_X509_num(certs)):
        x509 = crypto._lib.X509_dup(crypto._lib.sk_X509_value(certs, i))
        pycert = crypto.X509._from_raw_x509_ptr(x509)
        pycerts.append(pycert)
    if pycerts:
        return pycerts`

Reply:
I had a thread on the mailing list for the cryptography library about multiple certs. Last I checked, it was being worked on. I'll take a look at fixing this for enveloped certs as well.

Thanks!